subsys/bluetooth/host/id.c

```diff
@@ -1786,7 +1786,15 @@ int bt_id_set_scan_own_addr(bool active_scan, uint8_t *own_addr_type)
 
 	if (IS_ENABLED(CONFIG_BT_PRIVACY)) {
 		err = bt_id_set_private_addr(BT_ID_DEFAULT);
-		if (err) {
+		if (err == -EACCES &&
+				(atomic_test_bit(bt_dev.flags, BT_DEV_SCANNING) ||
+				 atomic_test_bit(bt_dev.flags, BT_DEV_INITIATING))) {
+			LOG_WRN("Ignoring failure to set random address while "
+				"scanning or initiating (%d)",
```

Comment:
Can't you compress the message so that you don't need to split it into two lines? (this makes it harder to grep for where the message came from). E.g. logging the error seems redundant since the branch is only entered if `err == -EACCESS`. Something like this perhaps:
```c
	LOG_WRN("Set random addr failure ignored in scan/init state");
```